tools/flakiness_checker.py

```diff
@@ -0,0 +1,87 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+""" Checks a given test for flakiness
+Takes the file name and function name of a test, as well as, optionally,
+the number of trials to run and the random seed to use
+"""
+
+import subprocess
+import sys
+import os
+import random
+import argparse
+
+
+DEFAULT_NUM_TRIALS = 500
+
+def run_test_trials(test_file, test_name, num_trials, seed):
+    test_path = test_file + ":" + test_name
+    
+    new_env = os.environ
+    new_env["MXNET_TEST_COUNT"] = str(num_trials)
+    if seed is None:
+        print("Using random seed")
+    else:
+        new_env["MXNET_TEST_SEED"] = seed
+
+    code = subprocess.call(["nosetests", "-s","--verbose",test_path], env = new_env)
+    print("nosetests completed with return code " + str(code))
+
+def find_test_path(test_file):
```

Comment:
Could you add some documentation? From the name alone it was not clear to me what this exactly does. Also, how is it going to handle duplicates and the fact that we import tests into other testfiles, e.g. test_operator_gpu?

Reply:
If you take a closer look at the code, you'll see that the file name and test name are both provided by the user. So if you would like to run the gpu version of test_sigmoid then you'll do \<path-to-test_operator_gpu.py\>:test_sigmoid or test_operator_gpu.test_sigmoid. The problem you're asking is probably "if we modified some tests in test_operator.py, how do we detect that we also need to run the gpu version of it?", which is designed to be handled by other components in the project (this is stated in the design doc too).